codeops-scripts/apply-github-settings.ps1

```diff
@@ -0,0 +1,198 @@
+#Requires -Modules @{ ModuleName="powershell-yaml"; ModuleVersion="0.4.2" }
+
+#
+# This script is responsible for ensuring all GitHub repositories are
+# configured as per a policy of defined repository settings.
+#
+# For example, enforcing a branch protection.
+# 
+
+[CmdletBinding()]
+param (
+    [string] $ConfigDirectory,
+    [switch] $WhatIf
+)
+$ErrorActionPreference = 'Stop'
+
+$here = Split-Path -Parent $PSCommandPath
+
+# Install other module dependencies
+$requiredModules = @(
+    @{ Name = "Endjin.CodeOps"; Version = "0.2.2" }
+)
+$requiredModules | ForEach-Object {
+    if ( !(Get-Module -ListAvailable $_ | Where-Object { $_.Version -eq $_.Version }) ) {
+        Install-Module -Name $_.Name `
+                       -RequiredVersion $_.Version `
+                       -AllowPrerelease:($_.Version -match '-') `
+                       -Scope CurrentUser `
+                       -Repository PSGallery `
+                       -Force
+    }
+    Import-Module -Name $_.Name -RequiredVersion $_.Version
+}
+
+
+#
+# Helper functions
+#
+function _getAllOrgReposWithDefaults {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $Org,
+
+        [Parameter(Mandatory = $true)]
+        [hashtable] $DefaultSettings
+    )
+
+    $allRepos = Invoke-GitHubRestMethod -Uri "https://api.github.com/orgs/$Org/repos?per_page=100" `
+                                        -AllPages | `
+                    Select-Object -ExpandProperty name
+    $reposToProcess = @{}
+    $allRepos | ForEach-Object {
+        # add each repo to the main processing list with default settings policy applied
+        # the defaults can be overridden by setting them in the YAML repo files
+        $reposToProcess += @{ "$_" = $DefaultSettings.Clone() } 
+    }
+
+    return $reposToProcess
+}
+function _getAllOrgs {
+    # returns all the orgs that we want included in this process
+    @(
+        "ais-dotnet"
```

Comment:
If this was driven by a yaml file - this could become more reusable for external folk

Reply:
Agreed. Originally I'd thought to derive the organisations from the existing YAML files, but parked it to make some progress.

Having got the other merging functionality working, I'll take another look.